### PR TITLE
New lint check: CheckDuplicateCase

### PR DESCRIPTION
**Pull request by Dominator008**
Check for duplicate case labels in a switch statement
Eg:
```javascript
switch (foo) {
  case 1:
  case 1:
}
```
This is normally an indication of a programmer error.

Inspired by ESLint (https://github.com/eslint/eslint/blob/master/lib/rules/no-duplicate-case.js)

[Review on Reviewable](https://reviewable.io/reviews/google/closure-compiler/1510)

### CODE REVIEW

**Comment by ChadKillingsworth** [comment]
Why defer this to a lint check? Shouldn't this be a suspicious code type warning? Where would this ever be valid?

**Comment by Dominator008** [comment]
@ChadKillingsworth I don't know. People might put weird logic in their code.

@MatrixFrog What do you think?

**Comment by ChadKillingsworth** [comment]
According to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch a duplicate label would never be matched - only the first case would be. It would only be available via fall-through.

I think the likelihood of this being an error is very high and a warning would be appropriate. As long as it is suppressible.

**Comment by Dominator008** [comment]
@ChadKillingsworth I don't have a preference. I can move this into CheckSuspiciousCode but there is a possibility that some internal code will be broken  :)

**Comment by MatrixFrog** [comment]
Let's keep it as a lint check so it won't break anyone, with a TODO to make it part of the suspicious-code group.

**Comment by MatrixFrog** [comment]
otherwise lgtm

**Comment by Dominator008** [comment]
@MatrixFrog Done.

**Comment by MatrixFrog** [comment]
I'll bring this in internally first since it will require a little bit of BUILD changes.

File: src/com/google/javascript/jscomp/lint/CheckDuplicateCase.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  @Override
+  public void hotSwapScript(Node scriptRoot, Node originalRoot) {
+    NodeTraversal.traverseEs6(compiler, scriptRoot, this);
+  }
+
+  @Override
+  public void visit(NodeTraversal t, Node n, Node parent) {
+    if (n.isSwitch()) {
+      Set<String> cases = new HashSet<>();
+      for (Node curr = n.getSecondChild(); curr != null; curr = curr.getNext()) {
```
Comment (Dominator008) [inline src/com/google/javascript/jscomp/lint/CheckDuplicateCase.java]: @MatrixFrog Just an FYI, I changed this from getFirstChild().getNext() to getSecondChild().

**Comment by MatrixFrog** [comment]
This is landed internally and will be MOE-pushed out soon (or it already was maybe?)

**Pull request closed without merging**